api/agent_client.go

```diff
@@ -185,29 +193,72 @@ func (c *AgentClient) GetScheduledJobs(ctx context.Context, afterCursor string,
 	}
 	defer resp.Body.Close()
 
-	result, retryAfter, err = decodeResponse[AgentScheduledJobs](resp)
+	result, retryAfter, err := decodeResponse[AgentScheduledJobs](resp)
 	if result != nil {
 		now := time.Now()
 		for _, j := range result.Jobs {
 			j.QueriedAt = now
-			if c.queue == defaultQueueKey {
-				// When we poll from default queue, we don't know the queue key, so in rest of the system queue="".
-				// The job might contain a queue key `agents: queue: default`, in that case it will cause mismatch in local
-				// job queue key "" vs our configuration queue key "default".
-				// So this is forcing them to be equal.
-				j.AgentQueryRules = agenttags.RemoveTag(j.AgentQueryRules, "queue")
-			} else {
-				// Ensure the job has the queue tag. We queried a queue-specific
-				// endpoint, but it may be the default queue, which doesn't require
-				// `agents: queue: ...`, so the queue tag might not be present.
-				j.AgentQueryRules = agenttags.SetTag(j.AgentQueryRules, "queue", c.queue)
-			}
+			j.AgentQueryRules = c.normaliseAgentQueryRules(j.AgentQueryRules)
 		}
 	}
+
 	return result, retryAfter, err
 }
 
-// AgentJob is enough info to run a job (includes command and env).
+func (c *AgentClient) normaliseAgentQueryRules(rules []string) []string {
+	if c.queue == defaultQueueKey {
+		// When we poll from default queue, we don't know the queue key, so in rest of the system queue="".
+		// The job might contain a queue key `agents: queue: default`, in that case it will cause mismatch in local
+		// job queue key "" vs our configuration queue key "default".
+		// So this is forcing them to be equal.
```

Comment:
> So this is forcing them to be equal.

just wanting to call out that, after refactor this code comment doesn't make sense without context. 

This particular code comment really only make sense to read around the code block that mutate job agent query rules.